### PR TITLE
Add logging markers

### LINKED ISSUE #154: Separate log messages by type
Currently all log messages are visible. For debug purposes create a better log manager that seperates messages by type?

### PR DESCRIPTION
**Pull request by ScribbleTAS**
Add a custom log4jconfig to the mod which allows markers, ansi-coloring and debug log level

## Markers
Markers allow you to filter log messages later  
To add a marker to log messages use:
```java
TASmod.logger.info(LoggerMarkers.Event, "Message");
```
This message can then later be disabled with 
```
-Dtasmod.marker.event=DENY
```
A new marker has to be added in the log4j.xml like this under `<Filters>`:
```xml
	<MarkerFilter marker="Event" onMatch="${sys:tasmod.marker.event:-ACCEPT}" onMismatch="NEUTRAL" />
```
## TASmod log level
The TASmod logger now has a seperate log level from the regular fabric one.  
It can be accessed with
```
-Dtasmod.log.level=INFO
```

## Misc

- Added more debug log messages and trace log messages
- Minor improvements to SavestateHandler
- Removed printing a stacktrace when no config file was found

## Todo
Markers to be added:

- [x] Events
- [x] Savestates
- [x] Networking //Add during packet rework!
- [x] Playback
- [x] Tickratechanger

Closes #154

### CODE REVIEW

File: src/main/java/com/minecrafttas/tasmod/events/client/EventDrawHotbar.java [hunk 1]
```diff
@@ -8,6 +8,7 @@ public interface EventDrawHotbar extends EventBase{
 	public void onDrawHotbar();
 	
 	public static void fireOnDrawHotbar() {
+		// No logging, because it is literally rendered every frame... This would spam the console even more
 		for (EventBase eventListener : EventListener.getEventListeners()) {
 			if(eventListener instanceof EventDrawHotbar) {
```
Comment (PancakeTAS) [inline src/main/java/com/minecrafttas/tasmod/events/client/EventDrawHotbar.java:13]: do we really need this comment

File: src/main/java/com/minecrafttas/tasmod/events/server/EventLoadstate.java [hunk 2]
```diff
@@ -3,13 +3,27 @@
...
+ *
+ */
 public interface EventLoadstate extends EventBase {
 	
+	/**
+	 * Fired when loading a savestate, before the savestate folder is copied
+	 * @param index
+	 * @param target
+	 * @param current
+	 */
```
Comment (PancakeTAS) [inline src/main/java/com/minecrafttas/tasmod/events/server/EventLoadstate.java]: you know you're meant to put something after the @param to tell me what the parameter does

File: src/main/java/com/minecrafttas/tasmod/events/server/EventSavestate.java [hunk 3]
```diff
@@ -3,13 +3,27 @@
...
+ *
+ */
 public interface EventSavestate extends EventBase {
 	
+	/**
+	 * Fired when saving a savestate, before the savestate folder is copied
+	 * @param index
+	 * @param target
+	 * @param current
+	 */
```
Comment (PancakeTAS) [inline src/main/java/com/minecrafttas/tasmod/events/server/EventSavestate.java]: same as before

File: src/main/java/com/minecrafttas/common/Configuration.java [hunk 4]
```diff
@@ -48,7 +48,7 @@ public Properties load() {
 			e.printStackTrace();
 			return null;
 		} catch (FileNotFoundException e) {
-			e.printStackTrace();
+			System.out.println("No config file found: "+file);
 			return null;
```
Comment (PancakeTAS) [inline src/main/java/com/minecrafttas/common/Configuration.java:52]: logging?

**Review by PancakeTAS — COMMENTED** [review]
just a few things to clear up

File: src/main/java/com/minecrafttas/common/Configuration.java (hunk 4 above)
Comment (ScribbleTAS) [reply src/main/java/com/minecrafttas/common/Configuration.java:52]: Plan is to make a cutom logger for the common package... But that is a story for another pr

File: src/main/java/com/minecrafttas/tasmod/events/client/EventDrawHotbar.java (hunk 1 above)
Comment (ScribbleTAS) [reply src/main/java/com/minecrafttas/tasmod/events/client/EventDrawHotbar.java:13]: I added a log statement for every event in TASmod, except that one. So yes

**Review by PancakeTAS — APPROVED** [review]

**Pull request merged**